packages/datadog-plugin-sharedb/src/index.js

```diff
@@ -0,0 +1,125 @@
+'use strict'
+
+/**
+ * @description The enum values in this map are not exposed from ShareDB, so the keys are hard-coded here.
+ * The values were derived from: https://github.com/share/sharedb/blob/master/lib/client/connection.js#L196
+ */
+const READABLE_ACTION_NAMES = {
+  hs: 'handshake',
+  qf: 'query-fetch',
+  qs: 'query-subscribe',
+  qu: 'query-unsubscribe',
+  bf: 'bulk-fetch',
+  bs: 'bulk-subscribe',
+  bu: 'bulk-unsubscribe',
+  f: 'fetch',
+  s: 'subscribe',
+  u: 'unsubscribe',
+  op: 'op',
+  nf: 'snapshot-fetch',
+  nt: 'snapshot-fetch-by-ts',
+  p: 'presence-broadcast',
+  pr: 'presence-request',
+  ps: 'presence-subscribe',
+  pu: 'presence-unsubscribe'
+}
+
+function getReadableActionName (action) {
+  const actionName = READABLE_ACTION_NAMES[action]
+  if (actionName === undefined) {
+    return action
+  }
+  return actionName
+}
+
+function getReadableResourceName (readableActionName, collection, query) {
+  if (collection) {
+    readableActionName += ' ' + collection
+  }
+  if (query) {
+    readableActionName += ' ' + JSON.stringify(sanitize(query))
+  }
+  return readableActionName
+}
+
+function sanitize (input) {
+  const output = {}
+
+  if (!isObject(input) || Buffer.isBuffer(input)) return '?'
+
+  for (const key in input) {
+    if (typeof input[key] === 'function') continue
+
+    output[key] = sanitize(input[key])
+  }
+
+  return output
+}
+
+function isObject (val) {
+  return typeof val === 'object' && val !== null && !(val instanceof Array)
+}
+
+function wrapCallback (config, tracer, request, span, done) {
+  return tracer.scope().bind((err, res) => {
+    if (err) {
+      span.addTags({
+        'error.type': err.name,
+        'error.msg': err.message,
+        'error.stack': err.stack
+      })
+    }
+
+    span.finish()
+
+    if (config.hooks && config.hooks.reply) {
+      config.hooks.reply(span, request, res)
```

Comment:
Hooks must be called before `span.finish()` is called since the span could be modified in the hook.